lib/common/strings.c

```diff
@@ -495,3 +495,49 @@ crm_strdup_printf(char const *format, ...)
     va_end(ap);
     return string;
 }
+
+/*!
+ * \brief Extract the name and value from an input string formatted as "name=value".
+ * If unable to extract them, they are returned as NULL.
+ *
+ * \param[in]  input The input string, likely from the command line
+ * \param[out] name  Everything before the first '=' in the input string
+ * \param[out] value Everything after the first '=' in the input string
+ *
+ * \return 2 if both name and value could be extracted, 1 if only one could, and
+ *         and error code otherwise
+ */
+int
+pcmk_scan_nvpair(const char *input, char **name, char **value) {
+#ifdef SSCANF_HAS_M
+    *name = NULL;
+    *value = NULL;
+    if (sscanf(input, "%m[^=]=%ms", name, value) <= 0) {
+        return -pcmk_err_bad_option;
+    }
+#else
+    char *sep = NULL;
+    *name = NULL;
+    *value = NULL;
+
+    sep = strstr(optarg, "=");
+    if (sep != NULL) {
+        *name = strndup(input, sep-input);
```

Comment:
It might be worthwhile to check the result here and below and return -ENOMEM if NULL, so the caller can distinguish a user error from a system error if desired